sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferStore.scala

```diff
@@ -128,10 +128,18 @@ abstract class RapidsBufferStore(
    * @param stream CUDA stream to use for copy or null
    * @return new buffer that was created
    */
-  def copyBuffer(buffer: RapidsBuffer, stream: Cuda.Stream): RapidsBufferBase = {
-    val newBuffer = createBuffer(buffer, stream)
-    buffers.add(newBuffer)
-    newBuffer
+  def copyBuffer(buffer: RapidsBuffer, memoryBuffer: MemoryBuffer, stream: Cuda.Stream)
+  : RapidsBufferBase = {
+    val newBuffer = createBuffer(buffer, memoryBuffer, stream)
+    try {
+      buffers.add(newBuffer)
+      catalog.registerNewBuffer(newBuffer)
+      newBuffer
+    } catch {
+      case e: IllegalStateException =>
```

Comment:
A bit of future-proofing here since we don't actually care what type of exception it is:
```suggestion
      case e: Exception =>
```

Reply:
I think the wrong exception was updated.  Instead of this being updated the exception we're suppressing later in this file was updated, see the other comment about that below.  This should still be updated since in this case we're not suppressing and no matter what the error is we want to free the buffer.

Reply:
Should be fixed now.